From b0d87abb373b053db71b9fe4a03f938703793483 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:17 -0400
Subject: [virt] kvm/ppc: book3s - move book3s_64_vio_hv.c into the main kernel binary

Message-id: <1410545655-205645-149-git-send-email-dzickus@redhat.com>
Patchwork-id: 94148
O-Subject: [RHEL7 PATCH 148/626] kvm: powerpc: book3s: move book3s_64_vio_hv.c into the main kernel binary
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 066212e02a8fb4ae797fae1d7cede5807ba398e1
Author: Paul Mackerras <paulus@samba.org>
Date:   Mon Oct 7 22:17:50 2013 +0530

    kvm: powerpc: book3s: move book3s_64_vio_hv.c into the main kernel binary

    Since the code in book3s_64_vio_hv.c is called from real mode with HV
    KVM, and therefore has to be built into the main kernel binary, this
    makes it always built-in rather than part of the KVM module.  It gets
    called from the KVM module by PR KVM, so this adds an EXPORT_SYMBOL_GPL().

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/Makefile b/arch/powerpc/kvm/Makefile
index 6646c95..104e8dc 100644
--- a/arch/powerpc/kvm/Makefile
+++ b/arch/powerpc/kvm/Makefile
@@ -53,32 +53,36 @@ kvm-e500mc-objs := \
  e500_emulate.o
 kvm-objs-$(CONFIG_KVM_E500MC) := $(kvm-e500mc-objs)
 
+kvm-book3s_64-builtin-objs-$(CONFIG_KVM_BOOK3S_64_HANDLER) := \
+ book3s_64_vio_hv.o
+
 kvm-book3s_64-objs-$(CONFIG_KVM_BOOK3S_64_PR) := \
  $(KVM)/coalesced_mmio.o \
  fpu.o \
  book3s_paired_singles.o \
  book3s_pr.o \
  book3s_pr_papr.o \
- book3s_64_vio_hv.o \
  book3s_emulate.o \
  book3s_interrupts.o \
  book3s_mmu_hpte.o \
  book3s_64_mmu_host.o \
  book3s_64_mmu.o \
  book3s_32_mmu.o
-kvm-book3s_64-builtin-objs-$(CONFIG_KVM_BOOK3S_64_PR) := \
+
+kvm-book3s_64-builtin-objs-$(CONFIG_KVM_BOOK3S_64_PR) += \
  book3s_rmhandlers.o
 
 kvm-book3s_64-objs-$(CONFIG_KVM_BOOK3S_64_HV) := \
  book3s_hv.o \
  book3s_hv_interrupts.o \
  book3s_64_mmu_hv.o
+
 kvm-book3s_64-builtin-xics-objs-$(CONFIG_KVM_XICS) := \
  book3s_hv_rm_xics.o
-kvm-book3s_64-builtin-objs-$(CONFIG_KVM_BOOK3S_64_HV) := \
+
+kvm-book3s_64-builtin-objs-$(CONFIG_KVM_BOOK3S_64_HV) += \
  book3s_hv_rmhandlers.o \
  book3s_hv_rm_mmu.o \
- book3s_64_vio_hv.o \
  book3s_hv_ras.o \
  book3s_hv_builtin.o \
  book3s_hv_cma.o \
diff --git a/arch/powerpc/kvm/book3s_64_vio_hv.c b/arch/powerpc/kvm/book3s_64_vio_hv.c
index 30c2f3b..2c25f54 100644
--- a/arch/powerpc/kvm/book3s_64_vio_hv.c
+++ b/arch/powerpc/kvm/book3s_64_vio_hv.c
@@ -74,3 +74,4 @@ long kvmppc_h_put_tce(struct kvm_vcpu *vcpu, unsigned long liobn,
  /* Didn't find the liobn, punt it to userspace */
  return H_TOO_HARD;
 }
+EXPORT_SYMBOL_GPL(kvmppc_h_put_tce);
-- 
1.7.1